docs/openid-auth.md

```diff
@@ -0,0 +1,152 @@
+- https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc
+
+## Prerequisites
+
+- Either Keycloak has been deployed or an existing Auth0/Okta account exist.
+  - Auth0 Signup: https://auth0.com/signup
+  - Okta Signup: https://developer.okta.com/signup/
+  - Getting started with Keycloak: https://www.keycloak.org/docs/latest/getting_started/
+- A value needs generated for `openid-secret`. This value will be used to encrypt the OpenID Connect session token.
+  - MacOS/Linux: `openssl rand -base64 25`
+  - Windows: https://activedirectoryfaq.com/2017/08/creating-individual-random-passwords/
+
+## Auth0 OpenID Connect Application Setup
+
+### Creating The Application
+
+1. Navigate to the Application management section of your Auth0 dashboard at `https://manage.auth0.com/dashboard/us/{{auth0_account_name}}/applications`
+2. Click the **_Create Application_** button in the top right of the page.
+3. Either provide a name for this application or use the default, then select **_Regular Web Application_** as the application type and click the blue **_Create_** button.
+   <kbd>
+   <img src="assets/openid-connect/auth0/create-application.png" />
+   </kbd>
+
+### Gather The Client ID
+
+1. Make note of the `Client ID` value. This value will be used in the Code-Server `openid-client-id` configuration variable.
+   <kbd>
+   <img src="assets/openid-connect/auth0/application-client-id.png" />
+   </kbd>
+
+### Update the application URLs
+
+1. Update the **_Allowed Callback URL_** and **_Allowed Logout URLs_** fields so that they point to the correct code-server endpoint.
+   <kbd>
+   <img src="assets/openid-connect/auth0/update-application.png" />
+   </kbd>
+
+### Example Auth0 Code-Server Configuration
+
+---
+
+**NOTE:** The `openid-issuer-base-url` should be set to the value of the **_Domain_** field of the Okta application.
+
+---
+
+```yml
+bind-addr: 127.0.0.1:8080
+cert: false
+
+auth: openid
+openid-issuer-base-url: "https://cdr.auth0.com"
+openid-client-id: "yg3IIz2alcikRqjqOsnRJABn8Uthfrtv"
+openid-base-url: "http://localhost:8080"
+openid-secret: "{prerequisites_openid-secret_value}"
+```
+
+## Okta OpenID Connect Application Setup
+
+### Creating The Application
+
+1. Navigate to the **_Add Application_** page of your Okta dashboard at `https://{{okta_account_name}}-admin.okta.com/admin/apps/add-app`
+2. Click the **_Create New App_** button located in the upper right of the page.
+3. Set the **_Platform_** to **_Web_** and **_Sign on method_** to **_OpenID Connect_**. Then click **_Create_**
+   <kbd>
+   <img src="assets/openid-connect/okta/create-application.png" />
+   </kbd>
+
+### Update The Application
+
+1. Update the **_Application name_** field with the desired name for this application.
+2. Update the **_Login redirect URIs_** and **_Logout redirect URIs (Optional)_** fields so that they point to the correct code-server endpoint then click **_Save_**
```

Comment:
```suggestion
2. Update the **Login redirect URIs** and **Logout redirect URIs (Optional)** fields so that they point to the correct code-server endpoint then click **Save**
```